src/object.c

```diff
@@ -377,6 +377,66 @@ void decrRefCount(robj *o) {
     }
 }
 
+void unusedStringObject(robj *o) {
```

Comment:
I think the term "unused" isn't good here.
maybe some adverb like "dismiss" or "dispose"?
or maybe "dontneed" to refer to our system level hack?

also, we need some big comment explaining this here, specifically why we don't bother to do that on some object types (small strings, etc)